colcon_bundle/verb/bundle.py

```diff
@@ -283,3 +289,38 @@ def _get_jobs(self, args, installers, decorators):
 
             jobs[pkg.name] = job
         return jobs
+
+    def _check_dependency_hash_match(self, bundle_base, decorators):
+
+        dependency_hash = {}
+
+        for decorator in decorators:
+            if not decorator.selected:
+                continue
+            pkg = decorator.descriptor
+            dependency_list = sorted(dependency.name for dependency in
+                                     pkg.dependencies['run'])
+            dependency_hash[pkg.name] = hashlib.sha256(
+                ' '.join(dependency_list).encode('utf-8')).hexdigest()
+
+        ordered_dependency_hash = OrderedDict(sorted(dependency_hash.items()))
+        current_hash_string = json.dumps(ordered_dependency_hash)
+        logger.debug('Hash for current dependencies: '
+                     '{current_hash_string}'.format_map(locals()))
+
+        dependency_hash_path = os.path.join(
+            bundle_base, 'dependency_hash.json')
+        dependency_tar_path = os.path.join(
+            bundle_base, 'dependencies.tar.gz')
+
+        if os.path.exists(dependency_hash_path) \
+                and os.path.exists(dependency_tar_path):
+            with open(dependency_hash_path, 'r') as f:
+                previous_hash_string = f.read()
+                if previous_hash_string == current_hash_string:
+                    return True
+
+        with open(dependency_hash_path, 'w') as f:
```

Comment:
This should be written after the install is successful, otherwise it's possible for `dependencies.tar.gz` to be out of sync with the dependencies in this file.